sherpa/astro/ui/tests/test_astro_ui_utils_simulation.py

```diff
@@ -817,3 +823,59 @@ def test_fake_pha_with_bgnd_data(clean_astro_ui, reset_seed):
 
     assert ui.get_source_plot().y == pytest.approx([2, 1, 0])
     assert ui.get_bkg_source_plot().y == pytest.approx([0, 0, 3])
+
+
+@pytest.mark.parametrize("grouped", [False, True])
+def test_fake_pha_grouping(grouped, clean_astro_ui):
+    """What happens with grouped data?"""
+
+    ui.dataspace1d(1, 5, dstype=ui.DataPHA)
+    ui.set_exposure(10)
+
+    grping = [1, -1, 1, -1, -1]
+    ui.set_grouping(1, grping)
+    ui.group(1)
+
+    egrid = np.arange(1, 7) * 0.5
+    elo = egrid[:-1]
+    ehi = egrid[1:]
+    rmf = create_delta_rmf(elo, ehi, e_min=elo, e_max=ehi)
+    ui.set_rmf(1, rmf)
+
+    assert ui.get_data().grouped
+
+    mdl = ui.create_model_component("polynom1d", "mdl")
+    ui.set_source(1, mdl)
+
+    ui.fake_pha(1, arf=None, rmf=None, exposure=10, grouped=grouped)
+
+    assert ui.get_grouping(1) == pytest.approx(grping)
+    assert ui.get_data().grouped is grouped
```

Comment:
I now this is a regression test. I wonder if we should write down this behaviour in a docstring somewhere. It's a choice we make, and I can find examples for or against it (e.g. grouping by "min_counts" would be different after a `fake_pha` call, while grouping by channel width would not). 
So, I agree not to touch whatever it's doing now, but if we codify with a test that this is what we want, should be bother to tell the user?

Reply:
I don't know what behavior you are asking about:

- is the grouping array retained
- is the grouped flag retained

I also don't know what you mean by "grouping by min_counts would be different" because that choice is made before the call - it's made when you call `group_counts` or `group_adapt_snr`. We don't "remember" the choice and re-apply it whenever the data or filter is changed [which is somehting we could think about doing but I would rather not] - we just have a grouping array and a quality array and how those arrays were created has been forgotten/lost.

Reply:
I'll make a PR onto your branch with a suggestion on what and where to write into the docs. That's probably easier than me suggesting things here inside a test which is definitely where the text is NOT going.

Reply:
https://github.com/DougBurke/sherpa/pull/4